src/Stats.h

```diff
@@ -17,17 +18,27 @@ class Stats
 public:
     Stats();
     void reset(StatsEnum stat);
-    void increment(StatsEnum stat, int increase = 1, float floatIncrease = 1.0);
-    void incrementFloat(StatsEnum stat, float increase = 1.0);
+    void resetAll();
+    void addDistance(int distance = 1);
+    void addTurn();
     void bindText(StatsEnum stat, QLineEdit* uiText);
-    void incrementEffectiveDistance(int distance);
-
+    void startRun();
+    void finishRun(bool reachedGoal);
+    void penalizeForReset();
 
 private:
     int intStatValues[NUMBER_OF_INT_STATS]; // total distance, turns
     float floatStatValues[NUMBER_OF_FLOAT_STATS]; // effective distance, score
-    QLineEdit* textField[NUMBER_OF_STATS];
+    QLineEdit* textField[NUMBER_OF_STATS] = {nullptr};
+    bool startedRun;
+    bool solved;
     void updateScore();
+    void increment(StatsEnum stat, int increase);
+    void increment(StatsEnum stat, float increase);
+    void setStat(StatsEnum stat, int value);
+    void setStat(StatsEnum stat, float value);
+    static float getEffectiveDistance(int distance);
+    float penalty = 0;
```

Comment:
Don't initialize here. Instead, use the member initialization list in the `.cpp` file.